security/identity-and-access-management/oracle-identity-governance/scheduled-task-samples/temporary-user-disable/files/src/com/oracle/sample/TemporaryUserDisable.java

```diff
@@ -0,0 +1,160 @@
+ * person obtaining a copy of this software, associated documentation and/or
+ * data (collectively the "Software"), free of charge and under any and all
+ * copyright rights in the Software, and any and all patent rights owned or
+ * freely licensable by each licensor hereunder covering either (i) the
+ * unmodified Software as contributed to or provided by such licensor, or (ii)
+ * the Larger Works (as defined below), to deal in both
+ *
+ * (a) the Software, and
+ *
+ * (b) any piece of software and/or hardware listed in the lrgrwrks.txt file if
+ * one is included with the Software each a "Larger Work" to which the Software
+ * is contributed by such licensors),
+ *
+ * without restriction, including without limitation the rights to copy, create
+ * derivative works of, display, perform, and distribute the Software and make,
+ * use, sell, offer for sale, import, export, have made, and have sold the
+ * Software and the Larger Work(s), and to sublicense the foregoing rights on
+ * either these or other terms.
+ *
+ * This license is subject to the following condition:
+ *
+ * The above copyright notice and either this complete permission notice or at a
+ * minimum a reference to the UPL must be included in all copies or substantial
+ * portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *******************************************************************************/
+
+package com.oracle.sample;
+
+import java.util.Calendar;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import oracle.iam.identity.usermgmt.api.UserManager;
+import oracle.iam.identity.usermgmt.api.UserManagerConstants;
+import oracle.iam.identity.usermgmt.vo.User;
+import oracle.iam.platform.Platform;
+import oracle.iam.platform.entitymgr.vo.SearchCriteria;
+import oracle.iam.scheduler.vo.TaskSupport;
+
+/**
+ * Sample scheduled task implementation that disables a user until a future
+ * enablement date is hit. The scheduled task will also clear the existing
+ * enablement date after enabling a user.
+ * 
+ * @author mgere-oracle
+ */
+public class TemporaryUserDisable extends TaskSupport {
+
+	private static final Logger logger = Logger.getLogger(TemporaryUserDisable.class.getName());
+
+	@Override
+	public void execute(HashMap arg0) throws Exception {
+		logger.log(Level.INFO, "The scheduled task run has started.");
```

Comment:
Missing input validation. What if arg0 is null?

Reply:
Hi Olaf.

A scheduled task's execute method is called by OIG's task scheduler, which always populates arg0 with the values of the parameters specified in the task scheduler definition (the xml part of this commit). These are populated by end users within OIG on task creation. As such arg0 can never be null and, at worst, can have an empty string set for the attribute name key in this particular case, in which case the code won't do anything useful (much like if the end user enters an invalid field name).

I can add a validation for arg0, but because of the above it would be entirely superfluous.

Reply:
Fixed.